pkg/bmc/access.go

```diff
@@ -44,6 +44,10 @@ type AccessDetails interface {
 	// (such as the kernel and ramdisk locations).
 	DriverInfo(bmcCreds Credentials) map[string]interface{}
 
+	// NodeProperties returns properties of a host, including the boot
+	// mode. This will depend on each interface
+	NodeProperties() map[string]interface{}
```

Comment:
I had originally thought we would return a map[string]interface{} because the driver would return data to be passed to Ironic directly. If we're not going to do that, let's define a real type to return so that the caller doesn't have to do the type assertion/conversion themselves.

Reply:
you are right, having an ipv6 on the bmc doesn't mean that the provision is going to be done through ipv6. How do you think we can sort that? Again we come to the need of exposing the parameter?